### PR TITLE
feature/tracing-data: Trace data as artifact in CI

### PR DESCRIPTION
**Pull request by patrickhuie19**
Decouples lifecycle of tempo + grafana UI containers from the lifecycle of smoke test runs by collecting trace data as an artifact from test runs.

### CODE REVIEW

File: .github/tracing/README.md [hunk 1]
```diff
@@ -1,5 +1,44 @@
...
 
-These config files are for an OTEL collector, grafana Tempo, and a grafana UI instance to run as containers on the same network.
+As part of the LOOP plugin effort, we've added distributed tracing to the core node. This is helpful for initial development and maintenance of LOOPs, but will also empower product teams building on top of core. 
+
+## Dev environment
+
+One way to generate traces locally today is with the OCR2 basic smoke test. 
+
+1. navigate to `.github/tracing/` and then run `docker compose --file local-smoke-docker-compose.yaml up`
+2. setup a local docker registry at `127.0.0.1:5000`
```
Comment (krehermann) [inline .github/tracing/README.md:10]: nit: link to some external doc about how to run the setup?

File: .github/tracing/README.md [hunk 2]
```diff
@@ -1,5 +1,44 @@
...
+## CI environment
+
+Another way to generate traces is by enabling traces for PRs. This will instrument traces for `TestOCRv2Basic` in the CI run. 
+
+1. Cut a PR in the core repo
+2. Add the `enable tracing` label to the PR
+3. Navigate to `Integration Tests / ETH Smoke Tests ocr2-plugins (pull_request)` details
+4. Navigate to the summary of the integration tests
+5. After the test completes, the generated trace data will be saved as an artifact, currently called `trace-data`
+6. Download this artifact to this directory
```
Comment (krehermann) [inline .github/tracing/README.md:28]: what is 'this' directory? put in the value so readers don't have to read both sections in order

File: .github/workflows/integration-tests.yml [hunk 3]
```diff
@@ -465,6 +431,10 @@ jobs:
           QA_AWS_REGION: ${{ secrets.QA_AWS_REGION }}
           QA_AWS_ROLE_TO_ASSUME: ${{ secrets.QA_AWS_ROLE_TO_ASSUME }}
           QA_KUBECONFIG: ${{ secrets.QA_KUBECONFIG }}
+      - name: Show Grafana Logs
```
Comment (krehermann) [inline .github/workflows/integration-tests.yml:434]: is this name correct? would except something about otel or trace in the name

File: .github/workflows/integration-tests.yml [hunk 4]
```diff
@@ -475,11 +445,22 @@ jobs:
           this-job-name: ETH Smoke Tests ${{ matrix.product.name }}${{ matrix.product.tag_suffix }}
           test-results-file: '{"testType":"go","filePath":"/tmp/gotest.log"}'
         continue-on-error: true
-      - name: Keep action running to view traces
+      - name: Wait for OTEL-Collector to flush buffered traces
         if: steps.check-label.outputs.trace == 'true' && matrix.product.name == 'ocr2' && matrix.product.tag_suffix == '-plugins'
         run: |
-          echo "Sleeping for $SLEEP_TIME seconds..."
-          sleep $SLEEP_TIME
+            sleep 60
```
Comment (krehermann) [inline .github/workflows/integration-tests.yml:451]: whats with the sleep? if needed, add a comment

**Review by jmank88 — DISMISSED** [review]

File: .github/workflows/integration-tests.yml [hunk 5]
```diff
@@ -475,11 +445,22 @@ jobs:
...
+      - name: Permissions on traces
+        if: steps.check-label.outputs.trace == 'true' && matrix.product.name == 'ocr2' && matrix.product.tag_suffix == '-plugins'
+        run: |
+            ls -l ./integration-tests/smoke/traces
+            # chmod -R 777 ./integration-tests/smoke/traces
```
Comment (krehermann) [inline .github/workflows/integration-tests.yml:456]: commented out?

Comment (patrickhuie19) [reply .github/workflows/integration-tests.yml:456]: obsoleted by changing the docker user for the otel-collector to match the user of the runner

**Review by krehermann — APPROVED** [review]

**Pull request merged**